app/src/main/java/eu/kanade/tachiyomi/widget/MangaSummaryView.kt

```diff
@@ -0,0 +1,188 @@
+package eu.kanade.tachiyomi.widget
+
+import android.animation.AnimatorSet
+import android.animation.ValueAnimator
+import android.content.Context
+import android.graphics.drawable.Animatable
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import android.widget.FrameLayout
+import androidx.annotation.AttrRes
+import androidx.annotation.StyleRes
+import androidx.constraintlayout.widget.ConstraintLayout
+import androidx.core.content.ContextCompat
+import androidx.core.view.doOnNextLayout
+import androidx.core.view.isVisible
+import androidx.core.view.updateLayoutParams
+import androidx.interpolator.view.animation.FastOutSlowInInterpolator
+import eu.kanade.tachiyomi.R
+import eu.kanade.tachiyomi.databinding.MangaSummaryBinding
+import eu.kanade.tachiyomi.util.system.animatorDurationScale
+import eu.kanade.tachiyomi.util.system.copyToClipboard
+import eu.kanade.tachiyomi.util.view.setChips
+import kotlin.math.roundToInt
+import kotlin.math.roundToLong
+
+class MangaSummaryView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    @AttrRes defStyleAttr: Int = 0,
+    @StyleRes defStyleRes: Int = 0
+) : FrameLayout(context, attrs, defStyleAttr, defStyleRes) {
+
+    private val binding = MangaSummaryBinding.inflate(LayoutInflater.from(context), this, true)
+
+    private var animatorSet: AnimatorSet? = null
+
+    private var recalculateHeights = false
+    private var descExpandedHeight = -1
+    private var descShrunkHeight = -1
+
+    var expanded = false
+        set(value) {
+            if (field != value) {
+                field = value
+                updateExpandState()
+            }
+        }
+
+    var description: CharSequence? = null
+        set(value) {
+            if (field != value) {
+                field = if (value.isNullOrBlank()) {
+                    context.getString(R.string.unknown)
+                } else {
+                    value
+                }
+                binding.descriptionText.text = field
+                recalculateHeights = true
+                doOnNextLayout {
+                    updateExpandState()
```

Comment:
You could check if the `lineCount` is less than or equal to 3 to then expand the description instead of doing `minLines`

Or instead of expanding you could try to hide the toggle and scrim, show the expanded pills, and disabled the click listeners that set the state to expanded

```suggestion
                    if (binding.descriptionText.lineCount <= 3) {
                        expanded = true
                    } else {
                        updateExpandState()
                    }
```

Reply:
> you could try to hide the toggle and scrim, show the expanded pills, and disabled the click listeners that set the state to expanded

I've tried this before but it's possible one manga have huge amount of tags, so I'm not entirely sure if it's wise to expand it by default.